notification/webhook/sender.go

```diff
@@ -0,0 +1,101 @@
+package webhook
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strconv"
+
+	"github.com/target/goalert/notification"
+)
+
+type Sender struct{}
+
+type WebhookAlert struct {
+	AlertID     int    `json:",omitempty"`
+	AlertType   string `json:",omitempty"`
+	Code        string `json:",omitempty"`
+	Summary     string `json:",omitempty"`
+	Details     string `json:",omitempty"`
+	ServiceID   string `json:",omitempty"`
+	ServiceName string `json:",omitempty"`
+	Count       int    `json:",omitempty"`
+	LogEntry    string `json:",omitempty"`
+}
+
+func NewSender(ctx context.Context) *Sender {
+	return &Sender{}
+}
+
+// Send will send an alert for the provided message type
+func (s *Sender) Send(ctx context.Context, msg notification.Message) (*notification.MessageStatus, error) {
+
+	postWithBody := func(body string) (*http.Response, error) {
+		req, err := http.NewRequest("POST", msg.Destination().Value, bytes.NewBufferString(body))
```

Comment:
If we move the switch statement out we can move this inline, additionally we can use the `[]byte` directly instead of casting to a string and also skip the `WithContext` by using `NewRequestWithContext`.

```suggestion
		req, err := http.NewRequestWithContext(ctx, "POST", msg.Destination().Value, bytes.NewBuffer(data))
```